### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by cristi-iacob**
- Using CMake instead of QMake;
- Modified Travis and Appveyor config files in order to run properly with CMake.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+find_library(LIBSMU_LIBRARIES smu)
+find_path(LIBSMU_INCLUDE_DIRS libsmu/libsmu.hpp)
+
+
+# generate version header
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -std=c++11")
+endif()
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG_BUILD")
```
Comment (dNechita) [inline CMakeLists.txt:85]: Why is this required?

Comment (AlexandraTrifan) [reply CMakeLists.txt:85]: I think we can remove it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+else()
+	find_library(LIBUSB_LIBRARIES usb-1.0)
+	find_path(LIBUSB_INCLUDE_DIRS libusb.h)
+endif()
+
+find_library(LIBSMU_LIBRARIES smu)
+find_path(LIBSMU_INCLUDE_DIRS libsmu/libsmu.hpp)
+
+
+# generate version header
```
Comment (dNechita) [inline CMakeLists.txt:81]: This comment seems misplaced.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+#List of warnings to be treated as errors
+target_compile_options(${PROJECT_NAME} PUBLIC
+	-Werror=return-type
+	-Werror=uninitialized
+	-Werror=init-self
+	-Werror=switch
+)
+
+if (WIN32 OR ENABLE_APPLICATION_BUNDLE)
+	message("================ Set target properties ")
```
Comment (dNechita) [inline CMakeLists.txt:204]: How is this message useful when someone runs cmake?

File: main.cpp [hunk 4]
```diff
@@ -17,7 +20,7 @@ int main(int argc, char *argv[])
 
     QLocale::setDefault(QLocale(QLocale::English, QLocale::UnitedStates));
 
-    init_signal_handlers(argv[0]);
+    /* init_signal_handlers(argv[0]); */
```
Comment (dNechita) [inline main.cpp:23]: If needed un-comment it, otherwise remove it. Leaving it like this rises questions like: "Why is this here?"

File: pixelpulse2.iss.cmakein [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #define AppName "Pixelpulse2"
-#define AppVersion "1.0.4"
+#define AppVersion "@GIT_VERSION@"
 #define AppPublisher "Analog Devices, Inc."
 #define AppURL "http://www.analog.com"
-#define AppExeName "Pixelpulse2.exe"
+#define AppExeName "pixelpulse2.exe"
```
Comment (dNechita) [inline pixelpulse2.iss.cmakein:5]: You can get the name of the executable from CMake.

File: utils/backtracing.h [hunk 6]
```diff
@@ -36,13 +38,26 @@ void windows_print_stacktrace(CONTEXT* context)
 
     memset(&frame, 0, sizeof(STACKFRAME));
 
-    frame.AddrPC.Offset         = context->Eip;
+//    frame.AddrPC.Offset         = context->Eip;
```
Comment (dNechita) [inline utils/backtracing.h:41]: This are not required anymore, right?
